controllers/articles.js

```diff
@@ -0,0 +1,195 @@
+import crypto from 'crypto';
+import models from '../models/index';
+
+const tempUser = '2dd9e22b-d19d-4501-a6b5-00bb4ebd9b3e'; // This line has to be deleted.
+const { article, User } = models;
+
+const articles = {
+
+  /*
+   * Deleting a post based to its slug.
+   *
+   * For directly publishing an article, status flag has to be passed
+   * into the request body object otherwise the article will be in drafted.
+   */
+  createAnArticle: async (req, res) => {
+    try {
+      const {
+        title, body, author, status, tagList,
+      } = req.body;
+      const flag = status === undefined ? 'Draft' : 'Published';
+      const tags = req.is('application/json') ? tagList : JSON.parse(tagList);
+
+      const slug = `${title.toLowerCase().split(' ').join('-').substring(0, 20)}${crypto.randomBytes(5).toString('hex')}`;
```

Comment:
I think this slug length is too short, most article's titles would be cut. I suggest you make it 30??

Reply:
@avpaul I have changed it to 40, then plus 10 random characters that makes it 50